src/src/Helpers/StringConverter.cpp

```diff
@@ -800,6 +800,47 @@ String tolerantParseStringKeepCase(const String& string, uint8_t indexFind, char
   return parseStringKeepCase(string, indexFind, separator, trimResult);
 }
 
+/*****************************************************************************
+ * handles: 0xXX,text,0xXX," more text ",0xXX starting from index 2 (1-based)
+ ****************************************************************************/
+String parseHexTextString(const String& argument, int index) {
+  String result;
+
+  // Ignore these characters when used as hex-byte separators (0x01ab 23-cd:45 -> 0x01,0xab,0x23,0xcd,0x45)
+  const String skipChars = F(" -:");
+
+  result.reserve(argument.length()); // longer than needed, most likely
+  int i      = index;
+  String arg = parseStringKeepCase(argument, i, ',', false);
+
+  while (!arg.isEmpty()) {
+    if ((arg.startsWith(F("0x")) || arg.startsWith(F("0X")))) {
+      size_t j = 2;
+
+      while (j < arg.length()) {
+        int hex = -1;
+
+        if (validIntFromString(concat(F("0x"), arg.substring(j, j + 2)), hex) && (hex > -1) && (hex < 256)) {
+          result += char(hex);
```

Comment:
A string with a `0` char does terminate at that point.

Reply:
Yes, I realized that too, so the check will be changed to `(hex > 0)` to fix that. No reason to include a 0-char, I think.